experimental/framework/tinylicious-client/src/test/TinyliciousClient.spec.ts

```diff
@@ -32,15 +149,193 @@ describe("TinyliciousClient", () => {
+    });
+
+    /**
+    * Scenario: test if an initialized TinyliciousClient can create container successfully when
+    * it is already initialized.
+    *
+    * Expected behavior: TinyliciousClient should use the existing instance instead of creating
+    * a new instance, then proceed to create a container.
+    */
+    it("initializing an initialized TinyliciousClient", async () => {
+        TinyliciousClient.init(clientConfig);
+        const containerConfig: TinyliciousContainerConfig = { id: documentId };
+        const schema: ContainerSchema = {
+            name: documentId,
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
+        const [container] = await TinyliciousClient.createContainer(containerConfig, schema);
+        await new Promise<void>((resolve, reject) => {
+            container.on("connected", () => {
+                resolve();
+            });
+        });
+        assert.notStrictEqual(container, undefined, "container is connected");
+    });
+
+    /**
+    * Scenario: test if the container can be created with an empty id in ContainerConfig
+    * and an empty name in ContainerSchema.
+    *
+    * Expected behavior: TinyliciousClient should throw an error
+    */
+    it("create container with empty id in containerConfig", async () => {
+        const containerConfig: TinyliciousContainerConfig = { id: "" };
+        const schema: ContainerSchema = {
+            name: "",
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
+        const containerAndServices = TinyliciousClient.createContainer(containerConfig, schema);
+        const errorFn = (error) => {
+            assert.notStrictEqual(error.message, undefined, "TinyliciousClient error is undefined");
+            return true;
+        };
+
+        await assert.rejects(
+            containerAndServices,
+            errorFn,
+            "TinyliciousClient can create container with empty ID",
+        );
+    });
+
+   /**
+    * Scenario: test if the initial objects that are passed in as parameter can be
+    * returned by the container.
+    *
+    * Expected behavior: initialObject ID is not empty.
+    */
+    it("use container to retrieve initial objects", async () => {
+        const containerConfig: TinyliciousContainerConfig = { id: documentId };
+        const schema: ContainerSchema = {
+            name: documentId,
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
         const [container] = await TinyliciousClient.createContainer(containerConfig, schema);
+        await new Promise<void>((resolve, reject) => {
+            container.on("connected", () => {
+                resolve();
+            });
+        });
+        const initialObjects = container.initialObjects;
+        const map1 = initialObjects.map1 as SharedMap;
+        assert.notStrictEqual(map1.id, "", "container can't connect with dynamic objects");
```

Comment:
If you're just trying to check that we can retrieve the object, I think checking that it's not undefined is probably better since having an empty id doesn't indicate failure in creating the map.  Realistically though, I'm not sure how much value checking this case is, and I think it would be better to include a check that the expected SharedMap functionality works (rather than just that it exists).  Also the comment should be initial objects instead of dynamic objects

Reply:
Yeah looking at the below two tests, I think those cover what this test is doing much more effectively since you rely on the objects (SharedMap/SharedDirectory) to function as expected rather that just exist.  Adding another test that checks that and that the same object loaded in two different containers mirror each other would be good too (essentially integrating your tests to create/add loadable objects dynamically and your test above to both create and get the same container)

Reply:
I separated the test into 2 different cases:
1. container to retrieve initial objects (initialObjects returned from `createContainer()` should mirror `getContainer()`)
2. can change initialObjects value (same object loaded in two different containers mirror each other even after object value is changed)